docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+> Also all requests made by our external monitoring are excluded from the whole SLO computation.
+
+
+# 3. Implementing SLO
+The SRE workbook contains examples how to implement SLO using Prometheus metrics.
+Initially we went this path and in some cases it can be the easiest way to start and eventually sufficient.
+But in some cases you can encounter issues that are hard to overcome and that was the reason we created the slo-exporter.
+If you want to read more about it take a look at [this series of blog posts we wrote](https://medium.com/@sklik.devops/our-journey-towards-slo-based-alerting-bd8bbe23c1d6).
+
+Ideally everyone in company should use the same approach how to compute or at least visualize SLOs. 
+In case slo-exporter is too much of a complexity for your trivial SLO computation,
+we advise you to at least try to stick to the metric naming used by slo-exporter and possibly using
+at least its alerting rules and dashboards.
+
+## Using Prometheus only
+This can be easy to implement, and you can find number of upstream
+projects that can help you with that such as [this SLO generator](https://promtools.dev/alerts/errors), 
+[Sloth](https://sloth.dev/) or [the Pyrra project](https://github.com/pyrra-dev/pyrra) having even a nice visualization.
+
+#### Advantages
+- No need for application changes, you can use already existing Prometheus metrics you already collect
+- No need for other infrastructure changes
+
+#### Disadvantages
+- Limited by cardinality because of the Prometheus architecture and purpose
+  (if needed for filtering of events or having more precise thresholds for some events)
+- For latency and quality SLI types need to use histograms and its buckets need to match the thresholds
+- Computing SLO based on high cardinality metrics over 4w can be enormously demanding for the Prometheus cluster
+
+## Using slo-exporter
+Slo exporter was built to solve more complex SLO setups and to unify SLO computation.
+It has support to compute SLO based on querying Prometheus data, tailing logs from file, receiving logs from Envoy proxy, following Kafka topic or reading logs from Elasticsearch.
+
+### 1. Choose source of the events
+#### Querying Prometheus
+This mode is mostly just for unification of the SLO computation so everyone uses the same tooling.
+It could be replaced by Prometheus recording rules and just hooking into the alerting by using the same metric naming schema as slo-exporter.
+It also does not support HA setup, so it's not advised to use this and avoid it if possible.
+Slo-exporter documentation can be found [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/prometheus_ingester.md) and complete example with whole configuration and kubernetes manifests
+[here](https://github.com/seznam/slo-exporter/tree/master/examples/prometheus).
+
+> Warning: in this setup, the slo-exporter cannot run in multiple replicas because it would lead to duplicating the observed events. Make sure to alert on that.
```

Comment:
```suggestion
> Warning: in this setup, it is important to not duplicate observed events - for example by running slo-exporter in more instances. In is adviced to run slo-exporter with prometheus ingester module as a single global replica or to shard the queries so that each slo-exporter instance handles different subset of events. Once you decide what is going to be your deployment model, create alerts to cover this.
```